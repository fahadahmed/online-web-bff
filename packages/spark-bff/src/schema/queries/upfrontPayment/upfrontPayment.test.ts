import { createTestClient, gql } from 'test/apolloTestUtils';

test('upfrontPayment query responds with correct mocked data', async () => {
  const { query } = createTestClient();

  const res = await query({
    query: gql`
      query UpfrontPaymentQuery($cartId: String!, $channel: CartChannel!) {
        upfrontPayment(cartId: $cartId, channel: $channel) {
          items {
            itemId
            title
            description
            price
            imagePath
          }
          monthlyIfpAmount
          requiredUpfrontAmount
        }
      }
    `,
    variables: {
      cartId: 'success-cart-id',
      channel: 'personalshop',
    },
  });
  expect(res.errors).toBeUndefined();
  expect(res.data).toMatchSnapshot();
});
