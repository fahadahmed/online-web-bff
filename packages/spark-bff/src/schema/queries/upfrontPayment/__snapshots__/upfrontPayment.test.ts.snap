// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`upfrontPayment query responds with correct mocked data 1`] = `
Object {
  "upfrontPayment": Object {
    "items": Array [
      Object {
        "description": "Blue",
        "imagePath": "/content/dam/spark/images/nui-graphics/static/EndlessPlanIcons/endless-plan-icon-child-1.png",
        "itemId": "512aab13-af1b-4a69-942d-083d6a23145f",
        "price": 52.75,
        "title": "iPhone 12 64GB",
      },
    ],
    "monthlyIfpAmount": 52.75,
    "requiredUpfrontAmount": 949.5,
  },
}
`;
